packages/seducer/src/getTypes.js

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable guard-for-in */
+/**
+ * Summary: From a given object with action functions declared returns a new object with the type names of those actions.
+ *
+ * @param {{[key]:(state, payload)=> nextState}}   actions  An object with {key: value} format that define an action of an state
+ *
+ * @return {{[key]: type}                          types    An object with all the names of the types inside
+ */
+
+export default function getTypes(actions) {
+  let o = {};
+  for (const action in actions) {
+    o = { ...o, [actions[action].name]: actions[action].name };
```

Comment:
By spreading `...o` on each iteration you create multiple redundant objects. There is no need for immutability here. 

```js
const types = {}
for (const action in actions) {
  types[actions[action].name] = actions[action]
}
return types;
```

Also, I guess it should be 

```
[actions[action].name]: actions[action]
```
not 
```
[actions[action].name]: actions[action].name
```

Also, why do we need to do `actions[action].name` ? Wouldn't just `action.name` be enough?

Reply:
I think this function is for generating an object with action types, but not an object with actions, so it has to be `[actions[action].name]: actions[action].name`

Reply:
Maybe this function needs a comment explaining and showing what is transformed to what. Or/and a unit-test.

Reply:
@allison-c is correct this allow you to do:

const actions = {
  add(),
  remove(id)
}

const [state, dispatch, action] = useSeducer()

dispatch(action.add)
dispatch(action.remove, id)